Mes 240 subscribe to state

vincent-re opened the pull request:
Fixes #
Update trackMessage
Update track
to use stored message and sdk filtering 

https://teamiflix.atlassian.net/browse/MES-119

Commit:
move track filtering from client-web to leanplum sdk
return an array of filtered messages
upate trackmessage to return an array of filtered messages

Commit:
publish

Commit:
publish

Commit:
publish and version bump

NikitaKurpas commented on package.json:
```diff
@@ -37,14 +37,12 @@
-  },
-  "optionalDependencies": {
+    "webpack": "^2.3.3",
     "fsevents": "^1.2.9"
```
Is this correct? I remember you had problems installing fsevents so that's why we moved it to optional

NikitaKurpas commented on src/Leanplum.js:
```diff
@@ -414,9 +426,13 @@ export default class Leanplum {
-      event,
-      params
+    events.publish( 'track', {
+      messages: Leanplum.getFilteredResults(
```
Why is the message sent to `track` channel, and not `track/message` here? Is there a difference in those channels?

lhoracek commented on src/Leanplum.js:
```diff
@@ -383,10 +383,22 @@ export default class Leanplum {
+  // TODO: need probably to add the event (aka exit, accept, cancel)
+  //  to build valid filter value: '.c23423938 exit'
+  //  either VarCache getMessageById or pass message or pass message params
+  static trackMessage( event, messageId) {
```
I don't see this whole function to be present on native SDKs and I can see only confusion when adding another name to already existent `track` and `advanceTo`. Leanplum knows only events and states (and so does business), so no need for new function from my point of view.

lhoracek reviewed (changes requested):
I would like to see the proprietary function to be removed and only `track` to be used.

i don

Commit:
use a single chanel for message event
remove unnecessary dependency
refactor trackMessage method
add an optional extra flag 'calledBy' to message event to allow extra filtering in clients-web

Commit:
version bump and publish
